src/compiler/checker.ts

```diff
@@ -31176,6 +31176,14 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     return contextualAwaitedType && getUnionType([contextualAwaitedType, createPromiseLikeType(contextualAwaitedType)]);
                 }
 
+                const symbol = getSymbolForExpression(node);
+
+                if (symbol && (symbol.flags & SymbolFlags.ObjectLiteral)) {
+                    return filterType(contextualReturnType, t => {
+                        return !getAwaitedTypeOfPromise(t);
+                    });

```

Comment:
Personally, I don't think this is the right fix. TS is a structural type at heart so this doesn't seem to be right - even if it's not overly practical for such types to contribute to contextual typing here... there is no type system rule that forbids it.

Completions are a separate thing and that could potentially be handled elsewhere (https://github.com/microsoft/TypeScript/blob/f37482cd16118409284e015409deaf14b416b4a0/src/services/completions.ts)

Reply:
Thank you very much for taking the time to review it. Oh There was a scope called "completion", i didn't know sorry.
I think this is related to autocomplete, so I think it's right to revise that part. I'll turn this PR to draft and research it more.

Reply:
I modified this inside the completion.

I'd appreciate it if you could give me your opinion when you have time!